test/SecTester.Scan.Tests/ScanTests.cs

```diff
@@ -1,53 +1,228 @@
+
+    // assert
+    act.Should().Throw<ArgumentNullException>().WithMessage("*options*");
+  }
+
+  [Fact]
+  public async Task Active_DefaultValue_ReturnsTrue()
+  {
+    // act
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions());
+
+    // assert
+    scan.Active.Should().BeTrue();
+  }
+
+  [Fact]
+  public async Task Done_DefaultValue_ReturnsTrue()
+  {
+    // act
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions());
+
+    // assert
+    scan.Done.Should().BeFalse();
+  }
+
+  [Fact]
+  public async Task Status_ReturnsState()
   {
     // arrange
-    var scans = Substitute.For<Scans>();
+    var scanState = new ScanState(ScanStatus.Running);
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions(PollingInterval: TimeSpan.Zero));
+
+    _scans.GetScan(ScanId).Returns(Task.FromResult(scanState));
 
     // act
-    var act = () => new Scan(scans, null!);
+    var result = await scan.Status().FirstAsync();
 
     // assert
-    act.Should().Throw<ArgumentNullException>().WithMessage("*options*");
+    result.Should().Be(scanState);
   }
 
   [Fact]
-  public void Active_DefaultValue_ReturnTrue()
+  public async Task Status_GivenCancelledToken_ThrowError()
   {
     // arrange
-    var scans = Substitute.For<Scans>();
+    using var cts = new CancellationTokenSource();
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions(PollingInterval: TimeSpan.Zero));
+
+    cts.Cancel();
 
     // act
-    using var scan = new Scan(scans, new ScanOptions());
+    var act = async () => await scan.Status(cts.Token).FirstAsync();
 
     // assert
-    scan.Active.Should().BeTrue();
+    await act.Should().ThrowAsync<OperationCanceledException>();
   }
 
   [Fact]
-  public void Done_DefaultValue_ReturnTrue()
+  public async Task Status_SwitchingDoneState_StopsYield()
   {
     // arrange
-    var scans = Substitute.For<Scans>();
+    var runningScanState = new ScanState(ScanStatus.Running);
+    var stoppedScanState = new ScanState(ScanStatus.Stopped);
+
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions(PollingInterval: TimeSpan.Zero));
+
+    _scans.GetScan(ScanId).Returns(Task.FromResult(runningScanState),
+      Task.FromResult(stoppedScanState));
 
     // act
-    using var scan = new Scan(scans, new ScanOptions());
+    var result = await scan.Status().LastOrDefaultAsync();
 
     // assert
-    scan.Done.Should().BeFalse();
+    result.Should().BeEquivalentTo(stoppedScanState);
+  }
+
+  [Fact]
+  public async Task Status_GivenQueuedStateConsequently_LogsConcurrencyWarning()
+  {
+    // arrange
+    var scanState = new ScanState(ScanStatus.Queued);
+
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions(PollingInterval: TimeSpan.Zero));
+
+    _scans.GetScan(ScanId).Returns(Task.FromResult(scanState),
+      Task.FromResult(scanState));
```

Comment:
BTW you can omit `Task.FromResult`

```suggestion
      scanState);
```